src/core/include/openvino/pass/pattern/op/optional.hpp

```diff
@@ -25,12 +25,12 @@ class OPENVINO_API Optional : public Pattern {
     /// \param patterns The pattern to match a graph.
     Optional(
         const std::vector<DiscreteTypeInfo>& type_infos,
-        const Output<Node>& pattern,
+        const OutputVector& patterns = {},
```

Comment:
We can set empty input as here. Just check the pattern 
![image](https://github.com/openvinotoolkit/openvino/assets/22008923/5745b73c-1965-4a39-ac65-9faa5c85acc7)

Reply:
![image](https://github.com/openvinotoolkit/openvino/assets/22008923/a7a4f397-ebbc-428a-ad9a-672ce16b9013)

Reply:
the question wasn't about empty inputs.
it was about > 1 inputs
please take a look at the initial comment

Reply:
@itikhono My comment was addressed to initial comment.
Let Me extend my comments by the details:
If we will use `pattern::optional<NodeType>` construction, we cannot use many inputs as it should be in the current state. As you can see in the code, I should use `Optional` without input sometimes as for `Stride` node. It's easiest approach to use empty output vector in the `Optional` constructor and add `pattern::optional<NodeType>()`. As I was done in this version.
But if we will use a constructor, we can have the mentioned case.

It's just one solution. But have another 2:
* to propagate using of OutputVector to `pattern::optional<NodeType>` to use optional as `wrap_type`. It means that we can use many inputs and specialize optional op per in pattern.
* to redefine constructors for 2 cases: Without inputs and With only input (as in master branch).

I think that best option is `to propagate using of OutputVector to `pattern::optional<NodeType>` to use optional as `wrap_type` because I saw some places where will be comfortable to use several input patterns

Reply:
can we simply introduce  a new ctor
 Optional(
        const std::vector<DiscreteTypeInfo>& type_infos, 
        const pattern::op::ValuePredicate& pred =
            [](const Output<Node>& output) {
                return true;
            })
without "patterns" argument in this PR?
after that (as part of another ticket in the next PR) we can try to extend it with multiple inputs?

Reply:
This is about this approach: `to redefine constructors for 2 cases: Without inputs and With only input (as in master branch).` :)